okhttp/src/main/kotlin/okhttp3/internal/tls/OkHostnameVerifier.kt

```diff
@@ -61,7 +60,7 @@ object OkHostnameVerifier : HostnameVerifier {
 
   /** Returns true if [certificate] matches [hostname]. */
   private fun verifyHostname(hostname: String, certificate: X509Certificate): Boolean {
-    val hostname = hostname.toLowerCase(Locale.US)
+    val hostname = DnsUtils.normalizeIA5String(hostname)
```

Comment:
should we use `hostname.toCanonicalHost()` instead?

Reply:
It's worse actually, causes us to fail on the tel example also.

```
okhttp3.internal.tls.HostnameVerifierTest > specialKInHostname FAILED
    org.junit.ComparisonFailure: expected:<[fals]e> but was:<[tru]e>
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at okhttp3.internal.tls.HostnameVerifierTest.specialKInHostname(HostnameVerifierTest.java:586)
```

Reply:
Maybe there is a fix where fix toCanonicalHost logic and then we use it, but maybe that's a good second pass PR once the test is in.

Reply:
What about just removing the `toLowerCase(Locale.US)` call completely? I’m trying to avoid having a bunch of code in here that doesn’t get exercised in practice.

Reply:
It's still public API

```


  @Test
  public void thatCatchesErrorsWithBadSession() {
    HostnameVerifier localVerifier = new OkHttpClient().hostnameVerifier();

    // Since this is public API, okhttp3.internal.tls.OkHostnameVerifier.verify is also
    assertThat(verifier).isInstanceOf(OkHostnameVerifier.class);

    SSLSession session = TlsUtil.localhost().sslContext().createSSLEngine().getSession();
    assertThat(localVerifier.verify("\uD83D\uDCA9.com", session)).isFalse();
  }
```